cmd/adapter/adapter.go

```diff
@@ -273,6 +273,12 @@ func (cmd *PrometheusAdapter) addResourceMetricsAPI(promClient prom.Client, stop
 		return err
 	}
 
+	metricsHandler, err := mprom.MetricsHandler()
+	if err != nil {
+		return err
+	}
+	server.GenericAPIServer.Handler.NonGoRestfulMux.HandleFunc("/metrics", metricsHandler)
```

Comment:
I haven't tested it yet myself, but I would expect this to cause a [panic](https://cs.opensource.google/go/go/+/refs/tags/go1.21.2:src/net/http/server.go;l=2517-2519) since `/metrics` has already a handler by default unless we set `EnabledMetrics = false` in the server config like in https://github.com/kubernetes-sigs/metrics-server/blob/f792d7022c270df795adda1de8bde7b1d4970c20/pkg/server/config.go#L74.